doc/RichTextArea/RichTextArea.md

```diff
@@ -0,0 +1,742 @@
+|MOVE_LEFT                 |Moves the caret one symbol to the left
+|MOVE_PARAGRAPH_END        |Moves the caret to the end of the current paragraph
+|MOVE_PARAGRAPH_START      |Moves the caret to the beginning of the current paragraph
+|MOVE_RIGHT                |Moves the caret one symbol to the right
+|MOVE_TO_DOCUMENT_END      |Moves the caret to end of the document
+|MOVE_TO_DOCUMENT_START    |Moves the caret to beginning of the document
+|MOVE_UP                   |Moves the caret one visual text line up
+|MOVE_WORD_LEFT            |Moves the caret one word left (previous word if LTR, next word if RTL)
+|MOVE_WORD_NEXT            |Moves the caret to the next word
+|MOVE_WORD_NEXT_END        |Moves the caret to the end of next word
+|MOVE_WORD_PREVIOUS        |Moves the caret to the previous word
+|MOVE_WORD_RIGHT           |Moves the caret one word right (next word if LTR, previous word if RTL)
+|PAGE_DOWN                 |Moves the caret one page down
+|PAGE_UP                   |Moves the caret one page up
+|PASTE                     |Inserts rich text from the clipboard
+|PASTE_PLAIN_TEXT          |Inserts plain text from the clipboard
+|REDO                      |Reverts the last undo operation
+|SELECT_ALL                |Selects all text in the document
+|SELECT_DOWN               |Selects text (or extends selection) from the current caret position one visual text line down
+|SELECT_LEFT               |Selects text (or extends selection) from the current position to one symbol to the left
+|SELECT_PAGE_DOWN          |Selects text (or extends selection) from the current position to one page down
+|SELECT_PAGE_UP            |Selects text (or extends selection) from the current position to one page up
+|SELECT_PARAGRAPH          |Selects text (or extends selection) of the current paragraph
+|SELECT_RIGHT              |Selects text (or extends selection) from the current position to one symbol to the right
+|SELECT_TO_DOCUMENT_END    |Selects text (or extends selection) from the current caret position to the end of document
+|SELECT_TO_DOCUMENT_START  |Selects text (or extends selection) from the current caret position to the start of document
+|SELECT_UP                 |Selects text (or extends selection) from the current caret position one visual text line up
+|SELECT_WORD               |Selects word at the caret position
+|SELECT_WORD_LEFT          |Extends selection to the previous word (LTR) or next word (RTL)
+|SELECT_WORD_NEXT          |Extends selection to the next word
+|SELECT_WORD_NEXT_END      |Extends selection to the end of next word
+|SELECT_WORD_PREVIOUS      |Extends selection to the previous word
+|SELECT_WORD_RIGHT         |Extends selection to the next word (LTR) or previous word (RTL)
+|UNDO                      |Undoes the last edit operation
+
+These functions and the key mappings can be customized using the control's **InputMap**.
+
+
+
+### CodeArea Control
```

Comment:
Just curious to know what languages CodeArea can support? Is it limited to Java or it can support syntax highlighting for other languages as well?

Reply:
So we provide an interface for plugging in any syntax highlighter.
We'll ship a simple java highlighter as an example - simple enough to be useful, but not connected to java compiler to be accurate to the letter.